From 65c465aa1353711ccd0d16edee2455757a7a641a Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Tue, 26 Nov 2024 22:54:21 +0000
Subject: [PATCH 6/7] Allow users to overwrite ld with environment variable in
 linker optimization script (#137331)

This should help in the case of cross compilation.

xref: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/261

Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/137331
Approved by: https://github.com/isuruf, https://github.com/seemethere
---
 tools/setup_helpers/generate_linker_script.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/setup_helpers/generate_linker_script.py b/tools/setup_helpers/generate_linker_script.py
index 11c397a9e5f..e66fc197062 100644
--- a/tools/setup_helpers/generate_linker_script.py
+++ b/tools/setup_helpers/generate_linker_script.py
@@ -1,3 +1,4 @@
+import os
 import subprocess
 
 
@@ -9,8 +10,8 @@ def gen_linker_script(
         prioritized_text = [
             line.replace("\n", "") for line in prioritized_text if line != "\n"
         ]
-
-    linker_script_lines = subprocess.check_output(["ld", "-verbose"], text=True).split(
+    ld = os.environ.get("LD", "ld")
+    linker_script_lines = subprocess.check_output([ld, "-verbose"], text=True).split(
         "\n"
     )
 
